app/models/bulk_assign_role.rb

```diff
@@ -0,0 +1,7 @@
+class BulkAssignRole < ApplicationRecord
```

Comment:
The model isn't necessary, and is potentially confusing. The error is happening because `admin` controllers have [a built-in authorization step](https://github.com/forem/forem/blob/main/app/controllers/admin/application_controller.rb#L27-L33) that tries to map the controller name to an object for use with the `InternalPolicy`.

You could skip the authorization, but this action really _ought_ to require an admin. The policy system does not actually require any ActiveRecord object, it's only used as the default in the `authorize_admin` method because it's relatively common. You can override that method without it:

```ruby
    def authorize_admin
      authorize nil, :access?, policy_class: InternalPolicy
    end
```

For what it's worth, that eventually winds it's way to [`administrative_access_to?`](https://github.com/forem/forem/blob/0b7545b2a4546f0d7637af7c23c32f07a9cccc2a/app/policies/authorizer.rb#L46-L56).

(Arguably better, but you could also refactor `authorization_resource` so that it returns a `nil` instead of throwing errors about missing constants, then you don't need to override `authorize_admin` at all.)

Reply:
I think that another way that we've dealt with case in the past where there is no resource to authorize is to [override the `authorization_resource` in the individual controller to return nil](https://github.com/forem/forem/blob/rt4914/bulk-assign-roles/app/controllers/admin/settings_controller.rb#L21).

Reply:
Thanks for the suggestions.

I have updated the code by removing the model and adding:

```
    # We need to override this method from Admin::ApplicationController since
    # there is no resource to authorize.
    def authorization_resource; end
```